### PR TITLE
build system: windows support + stylua style checker

### LINKED ISSUE #24: Better windows support (makefile)
Got this error while compiling the plugin with `make`

```
mkdir -p build
cc -O3 -Wall -Werror -fpic -shared src/fzf.c -o build/libfzf.so
process_begin: CreateProcess(NULL, cc -O3 -Wall -Werror -fpic -shared src/fzf.c -o build/libfzf.so, ...) failed.
make (e=2): The system cannot find the file specified.
make: *** [Makefile:8: build/libfzf.so] Error 2
```

make version:
```
GNU Make 4.3
Built for Windows32
Copyright (C) 1988-2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html> 
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

As you can see, I'm on Windows 10.
Same error on powershell, cmd and alacritty.

- Tried with version 3.81 of `make`, same error.

### PR DESCRIPTION
**Pull request by Conni2461**
Close #24

@luisiacc Can you try if this works now? Out of the box. So checkout branch and `make clean && make`

@ryanreno Can you check if OpenBSD still works after that. I need to force `CC=gcc` for windows. windows doesn't have `cc`. I only set it if the OS is windows but i still wanted to ask, if you can test it. Thanks :)

### CODE REVIEW

**Comment by luisiacc** [comment]
`make` works well now, `make clean` throws error(regular windows stuff), I had to manually remove `build` folder.
```
rm -rf build
process_begin: CreateProcess(NULL, rm -rf build, ...) failed.
make (e=2): The system cannot find the file specified.
make: *** [clean] Error 2
```

**Comment by Conni2461** [comment]
Thanks for testing. Wanna try and pull again?

I don't think i need to support everything but i think supporting `make` and `make clean` on windows makes sense :)

**Comment by ryanreno** [comment]
Latest commit 23a2f10 works as expected on OpenBSD. Thanks for checking in.

File: Makefile [hunk 1]
```diff
@@ -2,9 +2,11 @@ CFLAGS = -Wall -Werror -fpic
 COVERAGE ?=
 
 ifeq ($(OS),Windows_NT)
+    RM = cmd //C rmdir //Q //S
     CC = gcc
```
Comment (luisiacc) [inline Makefile]: no need to double //, that is only necesary for the other slash(`\\`) -> `RM = cmd /C rmdir /Q /S`

**Comment by luisiacc** [comment]
also, the `-p` folder is still being created, so when you `make` again, it warns that it was already created

File: Makefile (hunk 1 above)
Comment (Conni2461) [reply Makefile]: So you can't even blindly trust stackoverflow :rofl:

**Comment by Conni2461** [comment]
I'll remove the `-p` on windows and i add a minus in front of mkdir so it doesn't throw an error if it fails (build dir already exist). Give me 5min :)

**Comment by luisiacc** [comment]
Everythin works OK!

**Comment by Conni2461** [comment]
So we can merge it. Thanks for testing :)

**Pull request merged**